api/views.py

```diff
@@ -16,118 +15,121 @@
-	# callback = request.GET['callback']
-	versions = ds.get_all_versions_flat()
-	ret.content = json.dumps(versions)
-	# ret.content = callback+'('+ json.dumps(versions) +')'
-	return ret
+    ret = HttpResponse(content_type='application/json')
+    # callback = request.GET['callback']
+    versions = ds.get_all_versions_flat()
+    ret.content = json.dumps(versions)
+    # ret.content = callback+'('+ json.dumps(versions) +')'
+    return ret
 
 
 @ensure_csrf_cookie
 def get_current(request, q_name):
-	ret = HttpResponse(content_type='application/json')
-	quest = ds.get_current_questionnaire(q_name)
-	# callback = request.GET['callback']
-	ret.content = json.dumps(quest)
-	# ret.content = callback+'('+json.dumps(quest)+')'
-	return ret
+    ret = HttpResponse(content_type='application/json')
+    quest = ds.get_current_questionnaire(q_name)
+    # callback = request.GET['callback']
+    ret.content = json.dumps(quest)
+    # ret.content = callback+'('+json.dumps(quest)+')'
+    return ret
 
 
 @ensure_csrf_cookie
 def get_major_version(request, major, minor, q_name):
-	ret = HttpResponse(content_type='application/json')
-	quest = ds.get_specific_version(q_name, major, minor)
-	try:
-		perm = InstrumentAuth.objects.get(user=request.user,
-											 instrument__instrument_id=quest['instrument_id'],
-											 instrument__major_version=quest['version']['major']).to_dict()
-	except InstrumentAuth.DoesNotExist:
-		perm = {}
-	else:
-		perm = perm['permissions']
-
-	ret.content = json.dumps({
-		'questionnaire': quest,
-		'perms': perm
-	})
-	return ret
+    ret = HttpResponse(content_type='application/json')
+    quest = ds.get_specific_version(q_name, major, minor)
+    logger.debug('Hi FROM GET_MAJOR_VERSON')
+    try:
+        perm = InstrumentAuth.objects.get(user=request.user,
+                                          instrument__instrument_id=quest['instrument_id'],
+                                          instrument__major_version=quest['version']['major']).to_dict()
+    except InstrumentAuth.DoesNotExist:
+        perm = {}
+    else:
+        perm = perm['permissions']
+
+    ret.content = json.dumps({
+        'questionnaire': quest,
+        'perms': perm
+    })
+    return ret
+
 
 @ensure_csrf_cookie
 def get_interview_version(request):
-	ret = HttpResponse(content_type='application/json')
-	q_name = request.GET.get('q')
-	major = request.GET.get('major', 1)
-	minor = request.GET.get('minor', 'current')
-	logger.debug('Interview client requesting {}: {}'.format(q_name, major))
-	quest = ds.get_specific_version(q_name, major, minor)
-	callback = request.GET['callback']
-	ret.content = json.dumps(quest)
-	ret.content = callback+'('+ json.dumps(quest) +')'
-	return ret
+    ret = HttpResponse(content_type='application/json')
+    q_name = request.GET.get('q')
+    major = request.GET.get('major', 1)
+    minor = request.GET.get('minor', 'current')
+    logger.info('Interview client requesting {}: {}'.format(q_name, major))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You have a log statement that might use unsanitized input. Depending on the context, this could result in:

1. A log injection attack that breaks log integrity, forges log entries, or bypasses monitors that use the logs. To increase the security of your code, sanitize your inputs before logging them. [Learn more](https://cwe.mitre.org/data/definitions/117.html)

2. A sensitive information leak that exposes users&#39; credentials, private information, or identifying information to an attacker. To preserve privacy in your code, redact sensitive user information before logging it. [Learn more](https://cwe.mitre.org/data/definitions/532.html)

Source: The unsanitized input originates as the result of `get` assigned to `major` at edru_eda5-dash/api/views.py:74.